src/qudi/hardware/fast_adc/spectrum/si_main.py

```diff
@@ -0,0 +1,411 @@
+
+        constraints['possible_timebase_list'] = np.array([2**i for i in range(18)])
+        constraints['hardware_binwidth_list'] = (constraints['possible_timebase_list']) / 250e6 #maximum sampling rate 250 MHz
+
+        return constraints
+
+    def configure(self, binwidth_s, record_length_s, number_of_gates=1):
+        """
+        Configure the card parameters.
+        @param float binwidth_s: Length of a single time bin in the time trace
+                                  histogram in seconds.
+        @param float record_length_s: Total length of the timetrace/each single
+                                      gate in seconds.
+        @param int number_of_gates: optional, number of gates in the pulse
+                                    sequence. Ignore for not gated counter.
+
+        @return tuple(binwidth_s, gate_length_s, number_of_gates):
+                    binwidth_s: float the actual set binwidth in seconds
+                    gate_length_s: the actual set gate length in seconds
+                    number_of_gates: the number of gated, which are accepted
+        """
+        def set_dynamic_params(binwidth_s, record_length_s, number_of_gates):
+            self.ms.load_dynamic_params(binwidth_s, record_length_s, number_of_gates)
+            self.ms.calc_data_size_S(self.cs.acq_pre_trigs_S, self.cs.acq_post_trigs_S)
+            self.cs.calc_dynamic_cs(self.ms)
+            self.ms.calc_buf_params()
+            self.ms.calc_actual_length_s()
+            self.cs.get_buf_size_B(self.ms.seq_size_B, self.ms.reps_per_buf, self.ms.total_pulse)
+
+        def get_buffer_pointer():
+            self.ms.c_buf_ptr = self.cmd.cfg.return_c_buf_ptr()
+            if self.ms.gated:
+                self.ms.c_ts_buf_ptr = self.cmd.cfg.return_c_ts_buf_ptr()
+
+        def initialize_all():
+            self.cmd.process.input_ms(self.ms)
+            self.fetcher.init_buffer(self.ms)
+            self.data.initialize(self.ms, self.cs)
+            self.processor = DataProcessGated(self.data, self.fetcher) if self.ms.gated \
+                else DataProcessorUngated(self.data, self.fetcher)
+            self.commander.init(self.ms, self.processor)
+            self.loop_manager.input_ms(self.ms)
+
+        self.cmd.card.card_reset()
+        self.ms.gated = self.cs.gated
+        self.ms.num_channels = len(self.cs.ai_ch)
+
+        set_dynamic_params(binwidth_s, record_length_s, number_of_gates)
+
+        self.cmd.cfg.configure_all(self.cs)
+        get_buffer_pointer()
+
+        initialize_all()
+
+        return self.ms.binwidth_s, self.ms.record_length_s, self.ms.total_pulse
+
+    def get_status(self):
+        """
+        Receives the current status of the Fast Counter and outputs it as
+                    return value.
+
+                0 = unconfigured
+                1 = idle
+                2 = running
+                3 = paused
+                -1 = error state
+        """
+        return self._internal_status
```

Comment:
As far as I understood, the _internal_status is just a python variable, not associated with any response from the hardware itself. Thus during a measurement an error could occur with the card but when checking with get_status() one would still get the result that the card is running as _internal_status was set to this during the start of the measurement.
Maybe the interface allows for a status query of the card?

Reply:
It would be great if this error state is used when the error from the hardware is caught. I added this as TODO.